### PR TITLE
feat: Extend workflowstates

### PR DESCRIPTION
**Pull request by ThomasJunk**
Extend the given workflowstates with the backward paths from the `Admin`.

### CODE REVIEW

**Comment by s-l-teichmann** [comment]
No. E.g.  `{ReadWorkflow, DeleteWorkflow}:        {Editor, Reviewer},` Editors should not be allowed to move documents to the delete state. only Reviewers should be able to do so.

**Comment by ThomasJunk** [comment]
My bad, I took it from [main](https://github.com/ISDuBA/ISDuBA/blob/main/pkg/models/workflow.go#L47) without checking it.

File: client/src/lib/permissions.ts
Comment (s-l-teichmann) [inline client/src/lib/permissions.ts:1]: I think the matrix in the client should be machine generated to have a the single source of truth.

**Review by s-l-teichmann — CHANGES_REQUESTED** [review]

**Review by s-l-teichmann — APPROVED** [review]
From my side this done now. -> LGTM

**Review by JanHoefelmeyer — APPROVED** [review]
LGTM & works

**Pull request merged**